frontend/src/metabase/entities/model-indexes/utils.ts

```diff
@@ -8,15 +8,18 @@ import {
   isBoolean,
 } from "metabase-lib/types/utils/isa";
 import type FieldEntity from "metabase-lib/metadata/Field";
-import type Table from "metabase-lib/metadata/Table";
+import type Question from "metabase-lib/Question";
 
-const hasPk = (table?: Table) =>
-  !!table?.fields?.some(
-    (field: FieldEntity) => isPK(field) && isInteger(field),
-  );
+const hasSingleIntegerPk = (model?: Question) => {
+  const pkFields = model
+    ?.getResultMetadata()
+    ?.filter((field: Field) => isPK(field));
+
+  return pkFields?.length === 1 && isInteger(pkFields[0]);
```

Comment:
Why do we support only integer PKs?

Reply:
it's a backend limitation for v1. will likely be broadened in the future, but I think it adds a good deal of complexity there.